Makefile

```diff
@@ -1,7 +1,26 @@
-#!/usr/bin/env bash
+##########################
+#  OpenShift Marketplace - Build and Test  #
+##########################
+
+SHELL := /bin/bash
+PKG   := github.com/operator-framework/operator-marketplace/pkg
+MOCKS_DIR := ./pkg/mocks
+OPERATORSOURCE_MOCK_PKG := operatorsource_mocks
 
 all: osbs-build
 
 osbs-build:
 	# hack/build.sh
 	./tmp/build/build.sh
+
+mockgen := $(GOBIN)/mockgen
```

Comment:
Good point, what you described is exactly the default behavior of `go install`.  Here is an excerpt from https://golang.org/cmd/go/#hdr-GOPATH_environment_variable - `If the GOBIN environment variable is set, commands are installed to the directory it names instead of DIR/bin. GOBIN must be an absolute path.`

I can add the above explanation as comments.

Reply:
How does adding the comment resolve the issue? The problem here is that if `GOBIN` is not present, then `mockgen` execution will fail even if it is present in `GOPATH\bin`. In other words you should be following the default behavior of `go install`.

Reply:
If `GOBIN` is not defined, `generate-mocks` still works fine because when used `mockgen` is not referred via `$(mockgen)`.  So the default behavior of `go install` still applies.

Although, I can improve how we install `mockgen` using make.

Reply:
`make mockgen`: installs mockgen at proper location.
`make generate-mocks`: generates mocks using mockgen.